jscomp/main/melc_cli.ml

```diff
@@ -308,6 +309,25 @@ module Internal = struct
     in
     Arg.(value & opt_all string [] & info [ "bs-package-output" ] ~doc)
 
+  let bs_module_type =
+    let module_system_conv =
+      let parse m =
+        match Js_packages_info.module_system_of_string m with
+        | Some module_system -> Ok module_system
+        | None -> Error (`Msg (Format.asprintf "Invalid module system %s" m))
+      in
+      let print fmt ms =
+        Format.fprintf fmt "%s" (Js_packages_info.module_system_to_string ms)
+      in
+      Arg.conv ~docv:"method" (parse, print)
+    in
+    let doc = "Specify the module type for JS imports" in
+    let docv = "module-type" in
+    Arg.(
+      value
+      & opt (some module_system_conv) None
+      & info [ "bs-module-type" ] ~doc ~docv)
```

Comment:
a nit about naming: could this be `mel-module-type`, or even just `module-type`? As it's a new flag...

Reply:
good point. I think the `-bs-*` prefix was meant upstream as "new flags that aren't in OCaml". I'm not sure how to handle that here. 

I think it might be fine to call it `--module-type`, indede.